packages/engine/src/scene/components/VariantComponent.tsx

```diff
@@ -107,6 +111,18 @@ const VariantLevelReactor = React.memo(({ entity, level }: { level: number; enti
 
   const modelComponent = useOptionalComponent(entity, ModelComponent)
 
+  useEffect(() => {
+    //if the variant heuristic is set to Distance, add the DistanceFromCameraComponent
+    if (variantComponent.heuristic.value === 'DISTANCE') {
+      setComponent(entity, DistanceFromCameraComponent)
+      variantLevel.metadata['minDistance'].value === undefined && variantLevel.metadata['minDistance'].set(0)
+      variantLevel.metadata['maxDistance'].value === undefined && variantLevel.metadata['maxDistance'].set(0)
+    } else {
+      //otherwise, remove the DistanceFromCameraComponent
+      hasComponent(entity, DistanceFromCameraComponent) && removeComponent(entity, DistanceFromCameraComponent)
```

Comment:
why are we removing the DistanceFromCameraComponent when not in distance mode? it's needed for frustum culling to work properly in SceneObjectSystem

Reply:
Currently DistanceFromCameraComponent is only ever assigned to avatars on import. We're not using it for any Model assets, and probably shouldn't if it can be avoided